### ISSUE TITLE
Cherry-picking changes from other versions/forks

### ISSUE THREAD

**Issue by akien-mga** [MCMic/jumpnbump#3]
Last time I searched for a fork of jumpnbump, I've noticed that the versionning has been a bit chaotic, there is e.g. a version 1.55 that some distros provide (openSUSE, OpenMandriva) - did you look into it to see if it has changes that could be relevant to cherry-pick for this resurrected fork?

There are also some commits here http://repo.or.cz/w/jumpnbump.git that might be relevant.

I'll add more stuff here if I found other interesting repos.

**Comment by akien-mga** [comment]
Patches for choosing joysticks: https://www.unix-ag.uni-kl.de/~auerswal/jumpnbump/index.html

Additional levels (could be added to another repo or in the github downloads, just for the sake of having everything jump'n'bump-related in the same place): https://web.archive.org/web/20110830033044/http://jumpnbump.spaceteddy.net/
There's also this source for custom levels that you probably already know as you're an Arch guy :) https://aur.archlinux.org/packages/jumpnbump-levels/
And apparently the original website is still more or less working regarding level downloads: http://www.brainchilddesign.com/Bottom.php?Page=Games&SubPage=JnBLevels

**Comment by akien-mga** [comment]
Found the version "1.55": www.sourcefiles.org/Games/Action/Platform/jumpnbump-1.55.tar.gz
Actually it's labelled 1.50, and *might* have something interesting to look into according to the changelogs diff with this git repo:
```
$ diff -aur jumpnbump/ChangeLog jumpnbump-1.50/ChangeLog 
--- jumpnbump/ChangeLog 2016-01-06 18:26:08.965886160 +0100
+++ jumpnbump-1.50/ChangeLog    2004-06-16 21:13:36.000000000 +0200
@@ -1,9 +1,6 @@
-27 Jun 2004 - 1.51
-  - refactored code to make it a bit simpler
-  - Added AI by Ricardo Cruz
-
-30 Mar 2004 - 1.42
-  - added support for gzip and bzip2 packed levels.
+16 Jun 2004 - 1.50
+  - AI Bunnies
+  - Contributed level selectors for KDE and X
 
 21 Mar 2003 - 1.41
   - new scaling filter which works with 8bit graphics. Removed the old scaler
```
(i.e. level selectors for KDE and X).

**Comment by akien-mga** [comment]
openSUSE also has quite a few patches, some of them might be relevant if they haven't been added to this repo yet: https://build.opensuse.org/package/show/games/jumpnbump

**Comment by akien-mga** [comment]
> Actually it's labelled 1.50, and might have something interesting to look into according to the changelogs diff with this git repo:

I had a quick look by copying the contents of this "1.55" version on top of 9b10b24 (the last common commit I guess), but there doesn't seem to be much interesting, it's mainly the network code that was also added in 1.51, and there was a code refactoring on top of it so it's hard to compare versions after they have diverged.

**Comment by MCMic** [comment]
Choosing joysticks only seems useful if you got more than 4 joysticks plugged in, this seems highly unlikely.
For the other patches, could provide links for the one you think are worth something? (or even PR)

**Comment by akien-mga** [comment]
Yes I'll likely make PRs if some patches are worth integrating - I just wanted to start this issue first so that there's a trace of some research for future contributors that might wonder if we have this or that patch in the code base :)
